File: rb/features.rb
# Evervault SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module EvervaultFeatures
  def self.make_feature(name)
    case name
    when "base"
      EvervaultBaseFeature.new
    when "test"
      EvervaultTestFeature.new
    else
      EvervaultBaseFeature.new
    end
  end
end

File: py/evervault_sdk/features.py
# Evervault SDK feature factory

from evervault_sdk.feature.base_feature import EvervaultBaseFeature
from evervault_sdk.feature.test_feature import EvervaultTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EvervaultBaseFeature(),
        "test": lambda: EvervaultTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
